katdal/datasources.py

```diff
@@ -343,8 +342,31 @@ def __init__(self, telstate, chunk_store=None, timestamps=None,
         if chunk_store is None:
             data = None
         else:
-            data = ChunkStoreVisFlagsWeights(
-                chunk_store, telstate['chunk_name'], telstate['chunk_info'])
+            # Augment chunk_info with chunk name prefix of current stream
+            prefix = telstate['chunk_name']
+            chunk_info = telstate['chunk_info']
+            for info in chunk_info.values():
+                info['prefix'] = prefix
```

Comment:
Do you want to make this replace only if it doesn't exist? That would allow us to update viswriter/flagwriter to populate this chunk_info entry in future.